rfcs/0019-storing-client-data.md

```diff
@@ -310,19 +315,22 @@ Below is a diagram of how the components in this RFC will depend on each other:
 
 ### Archive Process
 
-To offload many database writes that has to be conducted by the daemon, we are going to delegate the writes to another process, which will be called an Archive Process. Whenever the `transition_frontier` and the `transaction_pool` makes an update, they will send the diff representing the update to the process. The diffs of the `transition_frontier` that the archive process will receive will be similar to `Transition_frontier` extension diffs. Currently, these diffs contain breadcrumbs and we cannot serialize breadcrumbs, so the diffs that clients will be receiving will be the following:
+To offload many database writes that has to be conducted by the daemon, we are going to delegate the writes to another process, which will be called the Archive Process. This means that some production versions of Coda, such as Codaslim, would not have the binary of this Archive Process. Whenever the `transition_frontier` and the `transaction_pool` makes an update, they will send the diff representing the update to the process. The archive process will receive the diffs from the daemon through GraphQL subscriptions. The diffs of the `transition_frontier` that the archive process will receive would be similar to `Transition_frontier` extension diffs. Currently, these diffs contain breadcrumbs and we cannot serialize breadcrumbs, so the diffs that clients will be receiving will be the following:
 
 ```ocaml
 module Transition_frontier_diff = struct
   type t =
     | Breadcrumb_added of {block:  (block, State_hash.t) With_hash.t; sender_receipt_chains_from_parent_ledger: Receipt_chain_hash.t Public_key.Compressed.Map.t}
     | Root_transitioned: of {new_: state_hash; garbage: state_hash.t list}
+    | Bootstrap of {lost_blocks : state_hash list}
```

Comment:
Do you know which blocks are lost in a bootstrap? What if you Ctrl-Ced coda and then restarted it later? What is ensuring the view of the transition frontier is consistent between the SQL database and the in memory daemon?

Reply:
Yes, we would know which blocks are lost during bootstrapping. Responded around line #333